### PR TITLE
fix(slider): resolve step & snap floating point precision

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #9684

## Summary

Resolve floating point precision issue when using `snap` and `step` properties.

repro: https://codepen.io/benelan/pen/jOjxbRg

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/slider/slider.tsx [hunk 1]
```diff
@@ -964,8 +965,14 @@ export class Slider
...
-    let snappedValue = Math.floor((value - min) / step) * step + min;
-    snappedValue = Math.min(Math.max(snappedValue, min), max);
+
+    // prevents floating point precision issues
+    const bigDecimalString = new BigDecimal(`${Math.round((value - min) / step)}`)
+      .multiply(`${step}`)
+      .add(`${min}`)
+      .toString();
+
+    let snappedValue = Math.min(Math.max(Number(bigDecimalString), min), max);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/slider/slider.tsx:975]: Sidebar: I think we can use the `clamp` util here.

**Review by jcfranco — APPROVED** [review]
🎚️🤏🔍👁️👄👁️

File: packages/calcite-components/src/components/slider/slider.e2e.ts [hunk 2]
```diff
@@ -204,6 +204,17 @@ describe("calcite-slider", () => {
     });
   });
 
+  it("step floating point precision", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      `<calcite-slider value="1.4" label-handles max="10" min="0.1" snap step="0.1"></calcite-slider>`,
```
Comment (jcfranco) [inline packages/calcite-components/src/components/slider/slider.e2e.ts:210]: Should've spotted this sooner, but can you use the `html` util here for consistent formatting?

File: packages/calcite-components/src/components/slider/slider.e2e.ts [hunk 3]
```diff
@@ -204,6 +204,17 @@ describe("calcite-slider", () => {
     });
   });
 
+  it.skip("step floating point precision", async () => {
```
Comment (driskull) [inline packages/calcite-components/src/components/slider/slider.e2e.ts:207]: how come this is being added but skipped?

Comment (benelan) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: Due to #10168. Should I add a comment?

Comment (benelan) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: The alternative was/is expecting the buggy value (`1.3`) and mentioning in the issue to update the test with the correct value (`1.4`). 

Would you prefer that option? Intentionally committing a bug into our tests would hurt, but I'm sure I could work through it after years of therapy.

Comment (driskull) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: Its kind of weird to be adding a new test that is skipped. Why add it at all?

Comment (driskull) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: Yeah maybe just add a comment about the bug/issue.

Comment (benelan) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: Yeah, I agree it's weird. I wanted to add a test for coverage of #9684, but didn't have time to finish #10168 before the release. The two issues are related to the point where I couldn't repro the floating point precision one without the rounding one occurring too. 

I think adding the test and skipping it is better than not adding it at all or expecting a buggy value. I'll add a comment, thanks.

Comment (driskull) [reply packages/calcite-components/src/components/slider/slider.e2e.ts:207]: Makes sense! Thanks

**Review by jcfranco — APPROVED** [review]
🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️
🛰️🚀🛰️🛰️🛰️🚀🛰️🛰️🚀🚀🛰️🛰️🚀🚀🚀🛰️🛰️🚀🚀🚀🛰️🚀🚀🚀🚀🛰️🚀🛰️
🛰️🚀🚀🛰️🛰️🚀🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🛰️🛰️🚀🛰️🛰️🛰️🛰️🚀🛰️
🛰️🚀🛰️🚀🛰️🚀🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🛰️🛰️🚀🚀🚀🛰️🛰️🚀🛰️
🛰️🚀🛰️🛰️🚀🚀🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🚀🛰️🛰️🛰️🛰️🚀🛰️🛰️🛰️🛰️🛰️🛰️
🛰️🚀🛰️🛰️🛰️🚀🛰️🛰️🚀🚀🛰️🛰️🚀🚀🚀🛰️🛰️🚀🚀🚀🛰️🚀🚀🚀🚀🛰️🚀🛰️
🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️🛰️

**Pull request merged**